src/query/storages/result_cache/src/common.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright 2023 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_exception::Result;
+use common_expression::Column;
+use common_expression::DataBlock;
+use common_io::prelude::deserialize_from_slice;
+use common_io::prelude::serialize_into_buf;
+use sha2::Digest;
+use sha2::Sha256;
+
+const RESULT_CACHE_PREFIX: &str = "_result_cache";
+
+#[inline(always)]
+pub(crate) fn gen_common_key(raw: &str) -> String {
+    format!("{:x}", Sha256::digest(raw))
+}
+
+#[inline(always)]
+pub(crate) fn gen_result_cache_meta_key(tenant: &str, key: &str) -> String {
+    format!("{RESULT_CACHE_PREFIX}/{tenant}/{key}")
+}
+
+#[inline(always)]
+pub(crate) fn gen_result_cache_dir(key: &str) -> String {
+    format!("{RESULT_CACHE_PREFIX}/{key}")
+}
+
+#[derive(serde::Serialize, serde::Deserialize)]
+pub(crate) struct ResultCacheValue {
+    /// The query SQL.
+    pub sql: String,
```

Comment:
It's original sql? maybe we should save original sql if not?

Reply:
It's the origin SQL. It's used for `system.query_result`.